partridge/geo.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+from shapely.geometry import Point
+import utm
+
+from partridge.utilities import pairwise
+
+EARTH_RADIUS_METERS = 6371009
+
+
+def bbox_from_point(lon, lat, meters):
+    easting, northing, zn, zl = utm.from_latlon(lat, lon)
+    point = Point(easting, northing)
+    bounds = point.buffer(meters).bounds
+    minx, miny, maxx, maxy = bounds
+    result = []
+    for e, n in ((minx, miny), (maxx, maxy)):
+        lat, lon = utm.to_latlon(e, n, zn, zl)
+        result.append(lon)
+        result.append(lat)
+    return tuple(result)
+
+
+def pairwise_great_circle_vec(lons, lats, earth_radius=EARTH_RADIUS_METERS):
+    assert len(lats) > 1
+    assert len(lats) == len(lons)
+    npoints = len(lats)
+    nsegs = npoints - 1
+    lon1 = np.zeros(nsegs)
+    lat1 = np.zeros(nsegs)
+    lon2 = np.zeros(nsegs)
+    lat2 = np.zeros(nsegs)
+    # There must be a better way to do this with numpy?
```

Comment:
How about:
```
def pairwise_great_circle_vec(lons, lats, earth_radius=6371009):
    assert len(lats) > 1
    assert len(lats) == len(lons)
    return great_circle_vec(lons[:-1], lats[:-1], lons[1:], lats[1:], earth_radius)
```

Reply:
This method is also more performant:

```
for i in range(10000):
    pairwise_great_circle_vec(lons, lats)
```
CPU times: user 670 ms, sys: 0 ns, total: 670 ms
Wall time: 673 ms

vs.
```
for i in range(10000):
    pairwise_great_circle_vec_alt(lons, lats)
```
CPU times: user 280 ms, sys: 0 ns, total: 280 ms
Wall time: 280 ms

Reply:
Another option would be to use something like `pairwise` (http://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_distances.html). One way to deal with distance issues in that case would be to recast all coordinate values in, say, an equal area meter projection and leverage the euclidean distance which, for the scale of a bus route, would result in comparable figures without risking degradation in accuracy.

Reply:
✅ `great_circle_vec(lons[:-1], lats[:-1], lons[1:], lats[1:], earth_radius)`

That looks great! Care to add that commit to the branch? I made you a collaborator.

Reply:
(no rush btw)